static/js/components/dashboard/CourseDescription.js

```diff
@@ -19,49 +18,69 @@ export default class CourseDescription extends React.Component {
     now: moment$Moment,
   };
 
-  render() {
-    const { course, now } = this.props;
-    let text = "", enrolled = "";
-    let firstRun: CourseRun = {};
-    if (course.runs.length > 0) {
-      firstRun = course.runs[0];
+  courseDate(label: string, date: Date): string {
+    let formattedDate = date.format(DASHBOARD_FORMAT);
+    return `${label}: ${formattedDate}`;
+  }
+
+  courseDateMessage(courseStatus: string, firstRun: CourseRun): string {
```

Comment:
i think it might be better if the logic to produce the course description text and the course tooltip text (from CoursePrice.js) was moved into CourseRow.js. both this file and CoursePrice.js have logic that is built around the value of `courseStatus`, and there is some similar logic with specific values of `courseStatus` (take `STATUS_VERIFIED_NOT_COMPLETED` for example - the logic to produce the description text and tooltip text is very similar). if you combined this code to produce the course description and tooltip text in CourseRow.js, you could pass the text as a prop to the necessary components. 

it's worth getting other opinions on this - @noisecapella @aliceriot what do you think of the idea above?